packages/cactus-plugin-keychain-azure-kv/src/main/typescript/plugin-keychain-azure-kv.ts

```diff
@@ -0,0 +1,245 @@
+  azurePassword: string;
+}
+
+export interface IPluginKeychainAzureKvOptions extends ICactusPluginOptions {
+  logLevel?: LogLevelDesc;
+  keychainId: string;
+  instanceId: string;
+  azureEndpoint: string;
+  azureCredentialType?: AzureCredentialType;
+  azureInMemoryCredentials?: IAzureInMemoryCredentials;
+  backend?: SecretClient;
+}
+
+export class PluginKeychainAzureKv
+  implements ICactusPlugin, IPluginWebService, IPluginKeychain {
+  public static readonly CLASS_NAME = "PluginKeychainAzureKv";
+
+  readonly vaultUrl: string;
+  private readonly log: Logger;
+  private readonly instanceId: string;
+  private endpoints: IWebServiceEndpoint[] | undefined;
+  private azureKvClient: SecretClient;
+
+  public get className() {
+    return PluginKeychainAzureKv.CLASS_NAME;
+  }
+
+  constructor(public readonly opts: IPluginKeychainAzureKvOptions) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(opts, `${fnTag} arg options`);
+    Checks.truthy(opts.keychainId, `${fnTag} arg options.keychainId`);
+    Checks.truthy(opts.instanceId, `${fnTag} options.instanceId`);
+    Checks.nonBlankString(opts.keychainId, `${fnTag} options.keychainId`);
+    Checks.nonBlankString(opts.azureEndpoint, `${fnTag} options.azureEndpoint`);
+    if (
+      opts.azureCredentialType &&
+      opts.azureInMemoryCredentials &&
+      opts.azureCredentialType == AzureCredentialType.InMemory
+    ) {
+      Checks.nonBlankString(
+        opts.azureInMemoryCredentials.azureTenantId,
+        `${fnTag} opts.azureInMemoryCredentials.azureTenantId`,
+      );
+      Checks.nonBlankString(
+        opts.azureInMemoryCredentials.azureClientId,
+        `${fnTag} opts.azureInMemoryCredentials.azureClientId`,
+      );
+      Checks.nonBlankString(
+        opts.azureInMemoryCredentials.azureUsername,
+        `${fnTag} opts.azureInMemoryCredentials.azureUsername`,
+      );
+      Checks.nonBlankString(
+        opts.azureInMemoryCredentials.azurePassword,
+        `${fnTag} opts.azureInMemoryCredentials.azurePassword`,
+      );
+    }
+
+    const level = this.opts.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+
+    this.instanceId = this.opts.instanceId;
+    this.vaultUrl = this.opts.azureEndpoint;
+
+    if (opts.backend) {
+      this.azureKvClient = opts.backend;
+    } else {
+      if (
+        (opts.azureCredentialType = AzureCredentialType.InMemory) &&
+        opts.azureInMemoryCredentials
+      ) {
+        const azureCredentials = new UsernamePasswordCredential(
+          opts.azureInMemoryCredentials.azureTenantId,
+          opts.azureInMemoryCredentials.azureClientId,
+          opts.azureInMemoryCredentials.azureUsername,
+          opts.azureInMemoryCredentials.azurePassword,
+        );
+        this.azureKvClient = new SecretClient(this.vaultUrl, azureCredentials);
+      } else {
+        const azureCredentials = new DefaultAzureCredential();
```

Comment:
The credentials can be of 6 types.
EnvironmentCredential
ManagedIdentityCredential
SharedTokenCacheCredential
VisualStudioCredential
VisualStudioCodeCredential
AzureCliCredential
InteractiveBrowserCredential

And none of them access path from a file.
Only the EnvironmentCredential type accepts one path to the certificate and private key pair in PEM or PFX format, which can authenticate the App Registration (AZURE_CLIENT_CERTIFICATE_PATH)
but these vars need to be set as environment vars and cant just point to a file, like in AWS
AZURE_TENANT_ID 
AZURE_CLIENT_ID 
AZURE_CLIENT_SECRET
AZURE_CLIENT_CERTIFICATE_PATH 
AZURE_USERNAME 
AZURE_PASSWORD

Reply:
No, I dont think so, we cannot pass in a file path for this one.
It can only take in the `DefaultAzureCredentialOptions` interface, which is 
![image](https://user-images.githubusercontent.com/56861721/122006395-69663980-cdd4-11eb-985f-e080d2f424bc.png)